### PR TITLE
add new RAI Utils package for common utilities shared across RAI packages

### PR DESCRIPTION
**Pull request by imatiach-msft**
## Description

Add new RAI Utils package for common utilities shared across RAI packages.

Specifically, the new package currently contains retry utilities for fetching datasets and making requests.  In the future, we anticipate more common utilities being added to this package.  For example, it's possible that the definition of cohorts may go into this light-weight package which has no dependencies.

## Areas changed

npm packages changed:

- [ ] responsibleai/causality
- [ ] responsibleai/core-ui
- [ ] responsibleai/counterfactuals
- [ ] responsibleai/dataset-explorer
- [ ] responsibleai/fairness
- [ ] responsibleai/interpret
- [ ] responsibleai/localization
- [ ] responsibleai/mlchartlib
- [ ] responsibleai/model-assessment

Python packages changed:

- [ ] raiwidgets
- [ ] responsibleai
- [ ] erroranalysis
- [ ] rai_core_flask

## Tests

- [ ] No new tests required.
- [ ] New tests for the added feature are part of this PR.
- [ ] I validated the changes manually.

## Documentation:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .github/workflows/CI-python.yml [hunk 1]
```diff
@@ -10,7 +10,8 @@ jobs:
   ci-python:
     strategy:
       matrix:
-        packageDirectory: ["rai_core_flask", "responsibleai", "erroranalysis"]
+        packageDirectory:
+          ["rai_core_flask", "responsibleai", "erroranalysis", "rai_utils"]
```
[the remark this answers is not in the document]
Comment (romanlutz) [reply .github/workflows/CI-python.yml:14]: I should add: By that I mean `raiutils` would be better.

File: rai_utils/setup.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    long_description = fh.read()
+
+# Use requirements.txt to set the install_requires
+with open('requirements.txt') as f:
+    install_requires = [line.strip() for line in f]
+
+setuptools.setup(
+    name=name,  # noqa: F821
+    version=version,  # noqa: F821
+    author="Roman Lutz, Ilya Matiach, Ke Xu",
```
Comment (romanlutz) [inline rai_utils/setup.py:21]: Very generous of you 🤣 I guess it's fine in any way you like it, you can definitely remove my name, too.

**Review by romanlutz — APPROVED** [review]
Love it!

File: .github/workflows/CI-python.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (imatiach-msft) [reply .github/workflows/CI-python.yml:14]: thanks, will update the package name!

[the remark this answers is not in the document]
Comment (imatiach-msft) [reply .github/workflows/CI-python.yml]: done!

**Pull request merged**